azure_li_services/runtime_config.py

```diff
@@ -101,6 +101,10 @@ def get_network_config(self):
         if self.config_data and 'networking' in self.config_data:
             return self.config_data['networking']
 
+    def get_machine_constraints(self):
+        if self.config_data and 'machine_constraints' in self.config_data:
+            return self.config_data['machine_constraints']
```

Comment:
Skip the 'and' condition simply use:

return self.config_data.get('machine_constraints', None)

Similar for the others